src/stream/ResponseCollector.jl

```diff
@@ -1,9 +1,90 @@
 # This file is a part of GraknClient.  License is MIT: https://github.com/Humans-of-Julia/GraknClient.jl/blob/main/LICENSE
 
-mutable struct ResponseCollector
-    collectors::Dict{UUID,Queue}
+struct ResponseCollector
+    collectors::Dict{Bytes,Channel{Transaction_Res_All}}
+    transact_result_channel::Channel{Proto.Transaction_Server}
+    access_lock::ReentrantLock
 end
 
+
+function ResponseCollector(transact_result_channel::Channel{Proto.Transaction_Server})
+    collectors = Dict{Bytes,Channel{Transaction_Res_All}}()
+    access_lock = ReentrantLock()
+    resp_col = ResponseCollector(collectors, transact_result_channel, access_lock)
+    res_task = @async response_worker(resp_col)
+    return resp_col
+end
+
+"""
+newId_result_channel(resp_collector::ResponsCollector, request::T) where {T<:Proto.ProtoType}
+    Function is meant to give back the result_channel in which the results for one request
+    will be collected.
+    Attention! Don't put a new Id manually on the ResponsCollector. It wouldn't be thread safe
+"""
+function newId_result_channel(resp_collector::ResponseCollector, req_id::Bytes)
```

Comment:
This looks like a good case to extend `Base.push!` function

Reply:
Yes. After merge I will change this.

Reply:
Done. Thanks for the advice.